src/components/sidebar/CallTreeSidebar.js

```diff
@@ -32,9 +40,56 @@ function SidebarDetail({ label, children }: SidebarDetailProps) {
   );
 }
 
-class CallTreeSidebar extends React.PureComponent<StateProps> {
+type JsModesProps = {|
+  +jsModes: { [mode: string]: Milliseconds },
+|};
```

Comment:
Why is this wrapped in an object with a single property?

And let's call it jsImplementation instead of mode here, too.

Reply:
This is how props work in React, if we want to be able to use it as a React stateless component